### PR TITLE
imp(test): turn on full backtraces and disable frame filtering

### LINKED ISSUE #3536: Turn on full backtraces using COLORBT_SHOW_HIDDEN=1
## Motivation

In #3489, the test failure was hard to diagnose, because `COLORBT_SHOW_HIDDEN=1` hid some of the function calls we needed to see.

### Specifications

> `COLORBT_SHOW_HIDDEN` env variable, disabling frame filtering

https://github.com/athre0z/color-backtrace/blob/master/CHANGELOG.md#v042-2020-05-19

## Related Issues

I wanted to fix this as part of #3489, but those CI configs are getting refactored in PR #3459

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

In #3489, the test failure was hard to diagnose, because `COLORBT_SHOW_HIDDEN=1` hid some of the function calls we needed to see.

Fixes https://github.com/ZcashFoundation/zebra/issues/3536

### Specifications

> `COLORBT_SHOW_HIDDEN` env variable, disabling frame filtering

https://github.com/athre0z/color-backtrace/blob/master/CHANGELOG.md#v042-2020-05-19

## Solution

- Disable frame filtering
- Add Rust full backtraces

## Review
Anyone can review this

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ on:
 env:
   CARGO_INCREMENTAL: 0
   RUST_BACKTRACE: full
+  COLORBT_SHOW_HIDDEN: '1'
```
Comment (teor2345) [inline .github/workflows/ci.yml]: I just checked the docs, and we need this too:
https://docs.rs/color-eyre/latest/color_eyre/#multiple-report-format-verbosity-levels
https://doc.rust-lang.org/std/backtrace/index.html#environment-variables

```suggestion
  RUST_BACKTRACE: full
  RUST_LIB_BACKTRACE: full
  COLORBT_SHOW_HIDDEN: '1'
```

File: .github/workflows/test-full-sync.yml [hunk 2]
```diff
@@ -26,6 +26,8 @@ on:
 env:
   CARGO_INCREMENTAL: '1'
   ZEBRA_SKIP_IPV6_TESTS: "1"
+  RUST_BACKTRACE: full
```
Comment (teor2345) [inline .github/workflows/test-full-sync.yml:28]:
```suggestion
  RUST_BACKTRACE: full
  RUST_LIB_BACKTRACE: full
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -29,6 +29,8 @@ on:
 env:
   CARGO_INCREMENTAL: '1'
   ZEBRA_SKIP_IPV6_TESTS: "1"
+  RUST_BACKTRACE: full
```
Comment (teor2345) [inline .github/workflows/test.yml:32]:
```suggestion
  RUST_BACKTRACE: full
  RUST_LIB_BACKTRACE: full
```

**Review by teor2345 — CHANGES_REQUESTED** [review]
Thanks!

There are about 7 other instances of `RUST_BACKTRACE` in Zebra, can you fix them all?

**Comment by teor2345** [comment]
I'm marking this as a low priority, because it only caused that one problem.

File: docker/zcash-params/Dockerfile [hunk 4]
```diff
@@ -28,9 +28,15 @@ ENV CARGO_HOME /app/.cargo/
...
 
-ARG RUST_BACKTRACE=1
+ARG RUST_BACKTRACE=0
 ENV RUST_BACKTRACE ${RUST_BACKTRACE}
 
+ARG RUST_LIB_BACKTRACE=0
+ENV RUST_LIB_BACKTRACE ${RUST_LIB_BACKTRACE}
+
+ARG COLORBT_SHOW_HIDDEN=0
+ENV COLORBT_SHOW_HIDDEN ${COLORBT_SHOW_HIDDEN}
```
Comment (teor2345) [inline docker/zcash-params/Dockerfile:38]: Optional: These env var commands don't quite match `docker/Dockerfile`, can we use the same style in both places?

**Review by teor2345 — APPROVED** [review]
Happy to approve as is, or anyone can approve after the minor fix.

**Comment by teor2345** [comment]
> Error: API rate limit exceeded for installation ID 1523599.

File: docker/zcash-params/Dockerfile (hunk 4 above)
Comment (gustavovalverde) [reply docker/zcash-params/Dockerfile:38]: I'll create a PR to standardize all Dockerfiles and to document a few steps.

**Pull request merged**